plugins/woocommerce-admin/client/products/fields/attributes/attributes.tsx

```diff
@@ -28,9 +29,33 @@ export const Attributes: React.FC< AttributesProps > = ( {
 
 	return (
 		<AttributeControl
-			attributeType="regular"
 			value={ attributes }
+			onAdd={ () => {
+				recordEvent( 'product_add_attributes_modal_add_button_click' );
+			} }
 			onChange={ handleChange }
+			onNewModalCancel={ () => {
+				recordEvent(
+					'product_add_attributes_modal_cancel_button_click'
+				);
+			} }
+			onNewModalOpen={ () => {
+				if ( ! attributes.length ) {
+					recordEvent( 'product_add_first_attribute_button_click' );
+					return;
+				}
+				recordEvent( 'product_add_attribute_button' );
+			} }
```

Comment:
Hmm... could we tie the recording of the `click` events more directly to the button clicks? If in the future, there was another way the modal was opened, we would be recording these `click` events when we shouldn't.

Reply:
I could be wrong here, but I would venture a guess that we do in fact want to track when the modal is opened and not necessarily just a click.  For example, even if a non-sighted user used the "Enter" key to open the modal, I would think we'd want to track the modal opening.

But cc @pmcpinto to clarify this.  If there's a chance that we add another way to open the "add new" modal in the future, would we want unique event names for this or share the event?

Reply:
I think having one event for click and another for the modal view would be better. Makes sense?

Reply:
> For example, even if a non-sighted user used the "Enter" key to open the modal, I would think we'd want to track the modal opening.

I don't know if we have any guidelines on this, but since that would trigger the button's click event (unless we are doing something custom with keyboard handling), it would be appropriate to log the `click` Tracks event.

I think it is important to have the Tracks event names correspond to what they are really tracking. As Pedro mentions, having two separate events would be ideal, if we really think we may want to analyze things based on both events.